Wasm/Engine.lean

```diff
@@ -143,45 +145,58 @@ mutual
       | _ => .error $ .local_with_given_id_missing i.index
 
   -- TODO: there's a StateT somewhere here. Just sayin'
+  -- TODO: we're not typechecking at all!
+  -- TODO: can locals change when executing nested instructions?
   partial def runOp (locals : List (Option String × Option StackEntry))
                     (stack : List StackEntry)
                     : Operation
                     → Except EngineErrors (List StackEntry)
     | .nop => pure stack
     | .const _t n => pure $ .num n :: stack
     | .add _t g0 g1 => do
-      let (stack', operand0) ← getSO locals stack g0
-      let (stack1, operand1) ← getSO locals stack' g1
+      let (stack', .num operand0) ← getSO locals stack g0
+      let (stack1, .num operand1) ← getSO locals stack' g1
       let res ← match operand0, operand1 with
-      | .num n0, .num n1 => match n0, n1 with
         | .i ⟨b0, i0⟩, .i ⟨_b1, i1⟩ => pure $ .num $ .i ⟨b0, i0 + i1⟩ -- TODO: check bitsize and overflow!
         | .f ⟨b0, f0⟩, .f ⟨_b1, f1⟩ => pure $ .num $ .f ⟨b0, f0 + f1⟩ -- TODO: check bitsize and overflow!
         | _, _ => throw .param_type_incompatible
       pure (res :: stack1)
+    | .block ts ops => do
+      -- TODO: currently, we only support simple [] → [valuetype*] blocks,
+      -- not type indices. For this reason, we start the block execution
+      -- with an empty stack to simulate 0-arity.
+      let go σ op := do runOp locals (←σ) op
```

Comment:
Lgtm except I can't read this \leftarrow notation again

Reply:
The left arrow notation is, AFAIU, a shorthand for `let sgm ← σ; runOp locals sgm op`